### ISSUE TITLE
Pleroma support

### ISSUE THREAD

**Issue by BooDoo** [BooDoo/CheapBotsTootSweet#14]
Especially to offer [beeping.town](https://beeping.town) as another bot-friendly alternative to [botsin.space](https://botsin.space)

KNOWN ISSUES:
Bridge 
  - **Statuses with `description` field (alt text) fail "silently"**
       using Mastodon FE on beeping.town, data is POSTed to `/media` with a "filename" attribute that seeds image description field; changing this field sends a PUT to `media/:id` with the `description` field. This multi-step workflow is [also documented for Mastodon](https://github.com/tootsuite/documentation/blob/master/Using-the-API/API.md#updating-a-media-attachment)

Frontend: No impact? (untested, lol I wish)
Backend: Untested

**Comment by BooDoo** [comment]
OK so the issue seems to be in the `mastodon-api` library which is defunct and should be replaced anyway.

The library submits any non-file params using url querystrings, even when POSTing. 

~line 186 in mastodon-api/lib/mastodon.js, the logic flow should be

```
if (finalParams.file) { .headers['Content-Type'] = "multipart/form-data"; .formData = finalParams) }
++ else if (method === "POST")  { .headers['Content-Type'] = "application/json"; .json=true; .body=finalParams }
else if (Object.keys(finalParams.length > 0) { .url = Mastodon.formEncodeParams(finalParams);
```

tested with `shitposter.club` as the Pleroma instance and this change worked as intended and still allowed successful posting to `botsin.space` using the same patched script.

**Comment by BooDoo** [comment]
ok general posting should be resolved by 
[b0ff014b45179896eb2d2f968ae0fbf6535f8b09] && [cb1737f]

~~TODO: Test for alt text error originally described.~~
Alt text is workflow is not causing problems, this looks good!

**Comment by BooDoo** [comment]
Haven't tested multi-image upload or spoiler text but I'm calling this an acceptable minimum-viable.
